Mage.Sets/src/mage/cards/t/TimotharBaronOfBats.java

```diff
@@ -0,0 +1,139 @@
+        anotherVampireFilter.add(SubType.VAMPIRE.getPredicate());
+    }
+
+    public TimotharBaronOfBats(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{4}{B}{B}");
+
+        this.addSuperType(SuperType.LEGENDARY);
+        this.addSubType(SubType.VAMPIRE);
+        this.addSubType(SubType.NOBLE);
+
+        this.power = new MageInt(4);
+        this.toughness = new MageInt(4);
+
+        // Ward—Discard a card.
+        this.addAbility(new WardAbility(new DiscardCardCost()));
+
+        // Whenever another nontoken Vampire you control dies, you may pay {1} and exile it.
+        // If you do, create a 1/1 black Bat creature token with flying.
+        // It gains “When this creature deals combat damage to a player,
+        // sacrifice it and return the exiled card to the battlefield tapped.”
+        this.addAbility(new DiesCreatureTriggeredAbility(
+                new DoIfCostPaid(new TimotharBaronOfBatsCreateBatEffect(), new ManaCostsImpl<>("{1}"))
+                        .setText("you may pay {1} and exile it. " +
+                                "If you do, create a 1/1 black Bat creature token with flying. " +
+                                "It gains \"When this creature deals combat damage to a player, " +
+                                "sacrifice it and return the exiled card to the battlefield tapped\"."),
+                false,
+                anotherVampireFilter,
+                true
+        ));
+    }
+
+    private TimotharBaronOfBats(final TimotharBaronOfBats card) { super(card); }
+
+    @Override
+    public Card copy() { return new TimotharBaronOfBats(this); }
+}
+
+class TimotharBaronOfBatsCreateBatEffect extends OneShotEffect {
+
+    TimotharBaronOfBatsCreateBatEffect() {
+        super(Outcome.Benefit);
+        staticText = "exile it. " +
+                "If you do, create a 1/1 black Bat creature token with flying. " +
+                "It gains \"When this creature deals combat damage to a player, " +
+                "sacrifice it and return the exiled card to the battlefield tapped\".";
+    }
+
+    private TimotharBaronOfBatsCreateBatEffect(final TimotharBaronOfBatsCreateBatEffect effect) { super(effect); }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player player = game.getPlayer(source.getControllerId());
+        if (player == null) { return false; }
+
+        Card card = game.getCard(targetPointer.getFirst(game, source));
+        if (card == null) { return false; }
+
+        player.moveCards(card, Zone.EXILED, source, game);
+
+        BatToken bat = new BatToken();
+        bat.addAbility(new DealsCombatDamageToAPlayerTriggeredAbility(
+                new TimotharBaronOfBatsReturnEffect(card, game),
+                false
+        ));
+
+        CreateTokenEffect createBatEffect = new CreateTokenEffect(bat, 1);
+        createBatEffect.setTargetPointer(new FixedTarget(card, game));
```

Comment:
Why are you setting a target pointer here?  `CreateTokenEffect` doesn't use a target.

Reply:
I thought I needed that to pass the information for which card to bring back, but I'm instead passing that info through the constructor of TimotharBaronOfBatsReturnEffect when I create the bat.

I will remove this